src/components/ellenTicTacToe.tsx

```diff
@@ -0,0 +1,240 @@
+  renderSquare(i: number) {
+    return (
+      <Square
+        value={this.props.squares[i]}
+        onClick={() => this.props.onClick(i)}
+        isWinning={this.props.winningArray[i]}
+      />
+    );
+  }
+  render() {
+    return (
+      <div>
+        <div className="board-row">
+          {this.renderSquare(0)}
+          {this.renderSquare(1)}
+          {this.renderSquare(2)}
+        </div>
+        <div className="board-row">
+          {this.renderSquare(3)}
+          {this.renderSquare(4)}
+          {this.renderSquare(5)}
+        </div>
+        <div className="board-row">
+          {this.renderSquare(6)}
+          {this.renderSquare(7)}
+          {this.renderSquare(8)}
+        </div>
+      </div>
+    );
+  }
+}
+interface GameState {
+  history: Array<{ squares: string[]; clickedSquare?: number[] }>;
+  stepNumber: number;
+  xIsNext: boolean;
+}
+// tslint:disable-next-line:max-classes-per-file
+export class Game extends React.Component<{}, GameState> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      history: [
+        {
+          squares: Array(9).fill(null),
+        },
+      ],
+      stepNumber: 0,
+      xIsNext: true,
+    };
+  }
+  handleClick(i: number) {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          clickedSquare: [Math.floor((i % 3) + 1), Math.floor(i / 3 + 1)],
+          squares,
+        },
+      ]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext,
+    });
+  }
+  jumpTo(step: number) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0,
+    });
+  }
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+    const moves = history.map(
+      (step: { squares: string[]; clickedSquare: number[] }, move: number) => {
+        const clickedSquare = step.clickedSquare;
+        const desc = move
+          ? `Go to move #${move} (Col: ${clickedSquare[0]} Row:${
+              clickedSquare[1]
+            })`
+          : 'Go to game start';
+        return (
+          <li key={move}>
+            <button
+              className={css(
+                {
+                  font: 'inherit',
+                  margin: 3,
+                  overflow: 'visible',
+                  textTransform: 'none',
+                },
+                { ':focus': { fontWeight: 'bold' } },
+              )}
+              onClick={() => this.jumpTo(move)}
+            >
+              {desc}
+            </button>
+          </li>
+        );
+      },
+    );
+    let status;
+    if (winner) {

```

Comment:
It'd be nice to make a method or arrow function for this: `const status = this.computeStatus()`.

Then each branch of the `if` can just return a value.